e2e/__tests__/watchDynamicRequires.test.ts

```diff
@@ -0,0 +1,17 @@
+/**
```

Comment:
Why is the test named `watch`?

Reply:
Because the original bug (and the most likely common place this bug is noticed) is with the "--watch" option.  The machinery involved in `--watch` and `--findRelatedTests` is mostly the same for this case, and it's easier to write the test with `--findRelatedTests`.

Reply:
Agree that the naming is confusing, maybe `dynamicRequireDependencies` or so

Reply:
I changed it for `dynamicRequireDependencies`